errors.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package mutex
+
+import (
+	"github.com/juju/errors"
+)
+
+var (
+	ErrTimeout   = errors.New("timeout acquiring mutex")
```

Comment:
It implies that we expect people to take specific action in response to this error, and I don't actually know what that would be. Tim?